### PR TITLE
Feature(OIDC,auth): optimization OIDC,add GetAllThirdPartyIdpAsync

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ClientRepository.cs [hunk 1]
```diff
@@ -46,6 +46,7 @@ public async Task<PaginatedList<Client>> GetPaginatedListAsync(int page, int pag
                     .Include(c => c.AllowedCorsOrigins)
                     .Include(c => c.ClientSecrets)
                     .Include(c => c.AllowedScopes)
+                    .AsSplitQuery()
```
Comment (doddgu) [inline src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ClientRepository.cs:49]: I don't think it needs split query, you just need control left join and inner join.

File: src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/IdentityResourceRepository.cs [hunk 2]
```diff
@@ -43,6 +43,7 @@ public async Task<List<IdentityResource>> GetListAsync()
         var identityResources = await _context.Set<IdentityResource>()
                                 .Include(idrs => idrs.UserClaims)
                                 .Include(idrs => idrs.Properties)
+                                .AsSplitQuery()
```
Comment (doddgu) [inline src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/IdentityResourceRepository.cs:46]: I don't think it needs split query, you just need control left join and inner join.

File: src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ApiScopeRepository.cs [hunk 3]
```diff
@@ -39,6 +39,7 @@ public async Task<PaginatedList<ApiScope>> GetPaginatedListAsync(int page, int p
                          .Include(apiScope => apiScope.UserClaims)
                          .ThenInclude(apiScope => apiScope.UserClaim)
                          .Include(apiScope => apiScope.Properties)
+                         .AsSplitQuery()
```
Comment (doddgu) [inline src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ApiScopeRepository.cs:42]: I don't think it needs split query, you just need control left join and inner join.

File: src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ApiResourceRepository.cs [hunk 4]
```diff
@@ -41,6 +41,7 @@ public async Task<PaginatedList<ApiResource>> GetPaginatedListAsync(int page, in
                         .Include(apiResource => apiResource.Properties)
                         .Include(apiResource => apiResource.ApiScopes)
                         .ThenInclude(apiScope => apiScope.ApiScope)
+                        .AsSplitQuery()
```
Comment (doddgu) [inline src/Contrib/Authentication/OpenIdConnect/Masa.Contrib.Authentication.OpenIdConnect.EFCore/Repositories/ApiResourceRepository.cs:44]: I don't think it needs split query, you just need control left join and inner join.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test.csproj [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="6.0.8" />
```
Comment (zhenlei520) [inline src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test.csproj:11]: using `Directory.Build.props`

File: src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/TestDbContext.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+namespace Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test;
+
+public class TestDbContext : IsolationDbContext
+{
+    public TestDbContext(MasaDbContextOptions<TestDbContext> options) : base(options)
+    {
+
+    }
+
+    public DbSet<UserClaim> UserClaims { get; set; }
```
Comment (zhenlei520) [inline src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/TestDbContext.cs:13]: Properties are above the constructor

File: src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/_Imports.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+global using Microsoft.VisualStudio.TestTools.UnitTesting;
+global using Moq;
```
Comment (zhenlei520) [inline src/Contrib/Authentication/Tests/Masa.Contrib.Authentication.OpenIdConnect.EFCore.Test/_Imports.cs:5]: global using sort

**Review by doddgu — APPROVED** [review]
LGTM

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Review by zhenlei520 — APPROVED** [review]
LGTM

**Pull request merged**